src/main/kotlin/no/nav/familie/baks/mottak/integrasjoner/BaksVersjonertSøknadClient.kt

```diff
@@ -0,0 +1,56 @@
+package no.nav.familie.baks.mottak.integrasjoner
+
+import no.nav.familie.http.client.AbstractRestClient
+import no.nav.familie.kontrakter.ba.søknad.VersjonertBarnetrygdSøknad
+import no.nav.familie.kontrakter.felles.Ressurs
+import no.nav.familie.kontrakter.ks.søknad.VersjonertKontantstøtteSøknad
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.retry.annotation.Backoff
+import org.springframework.retry.annotation.Retryable
+import org.springframework.stereotype.Component
+import org.springframework.web.client.RestOperations
+import java.net.URI
+
+private val logger = LoggerFactory.getLogger(BaksVersjonertSøknadClient::class.java)
+
+@Component
+class BaksVersjonertSøknadClient
+    @Autowired
+    constructor(
+        @param:Value("\${FAMILIE_INTEGRASJONER_API_URL}")
+        private val integrasjonerServiceUri: URI,
+        @Qualifier("clientCredentials") restOperations: RestOperations,
+    ) : AbstractRestClient(restOperations, "integrasjon.saf") {
```

Comment:
Er det nødvendig med `@Autowired` og `constructor` her? Kunne ikke dette sett slik ut:
```suggestion
@Component
class BaksVersjonertSøknadClient(
        @param:Value("\${FAMILIE_INTEGRASJONER_API_URL}")
        private val integrasjonerServiceUri: URI,
        @Qualifier("clientCredentials") restOperations: RestOperations,
    ) : AbstractRestClient(restOperations, "integrasjon.saf") {
```

Tror også vi ikke "bryr" oss om hvilken tjeneste `familie-integrasjoner` snakker med i forbindelse med dette kallet. Ser ut som at vi stort sett bruker `AbstractRestClient(restOperations, "integrasjon")` de gangene vi setter opp ny client mot integrasjoner.

Reply:
Hmm nei, det gjør man ikke. Dette var inspirert av de andre klientene, skulle vi kanskje ryddet opp i dem også?

Ditto med strengen: "integrasjon.saf", den plukket jeg fra JournalpostClient, og så er det litt vekslende hva man putter der.

Jeg fikser det i hvert fall for mitt nå, så kan vi se på resten en annen gang.